### DISCUSSION TITLE
What is the relationship between Avalonia and Skia?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#13720] [Q&A] [answered]
Recently, I was chatting with a friend about Avalonia. My friend mentioned that Flutter has abandoned Skia due to Skia’s performance issues and is using the new Impeller instead. He believes that since Avalonia is based on Skia, Avalonia will also have the same performance problems, especially on the mobile phone.

But another friend of mine also mentioned that it seems that SkiaSharp and Skia are not exactly the same. Mono has modified Skia so that Skia can take advantage of the high performance of .Net. And he said that it seems that the SkiaSharp used by Avalonia is different from the SkiaSharp of Mono. It seems that the Avalonia team has made further modifications to SkiaSharp to make it perform better. But he also said that this was a rumor, and there seemed to be no such introduction or description on the Internet.

I would like to know whether Skia is the same as the Skia currently used by Avalonia except that it uses a different platform? In addition, will Avalonia also encounter the same performance problems that Flutter Skia encountered on the mobile phone? Thanks

**Comment by timunie** [comment, marked as answer]
well, I think you know the performance of Avalonia? For me it is good enough even on mobile, but still we do much work to improve performance further.

SkiaSharp is becomming faster by v3, which is in the oven, see #12604 

Last but not least our rendering system is pluggable. This means if there is an issue or someone has way to much time, we can techically add another rendering backend to impeller or whatever. So we are not tied to Skia. For example we had (or is it still there?) a D2D1 render backend on Windows :-)

**Reply by kyuranger** [reply]
I have only worked on desktop and embedded Avalonia, but never on mobile. From my personal point of view, the desktop performance is not as good as WPF (simply judging from the response speed of the interface to the naked eye), but Avalonia definitely does not have any performance problems.
In the future, I plan to use Avalonia to develop mobile programs, but I saw an article on the Internet saying that Flutter had serious lagging problems when developing iOS programs, so Google later changed the rendering engine to Impeller. For this reason, I talked about it with my friend.
Finally, thank you to the Avalonia team for their answers and continued efforts on the project. Looking forward to the launch of SkiaSharp V3 version!

**Comment by rabbitism** [comment]
https://github.com/AvaloniaUI/Avalonia/pull/11914

I think avalonia team is also experimenting on direct metal support.

**Reply by kyuranger** [reply]
Thanks master dong bin.

**Comment by sgf** [comment]
Remember SkiaSharp is just a wrapper
Unfortunately, there is no pure C# alternative in the .net ecosystem.
A high performance, cross-platform 2D vector graphics library developed by .net is so scarce.

**Reply by kyuranger** [reply]
As the .Net version is upgraded, the performance of .Net will also be improved. Will the performance of SkiaSharp also be improved? Or will there be no change?

**Reply by sgf** [reply]
> As the .Net version is upgraded, the performance of .Net will also be improved. Will the performance of SkiaSharp also be improved? Or will there be no change?

Previously, due to the emergence of function pointers, the performance of PInvoke has been improved. (Reducing the resource consumption of the interaction between C# and C libraries)
But in the future, unless Skia itself is significantly improved, there will be almost no visible improvement.
